kahuna/public/js/preview/image.html

```diff
@@ -75,18 +75,20 @@
                 {{::ctrl.image.data.uploadTime | date:'HH:mm'}}
             </span>
 
-            <span class="bottom-bar__meta-item titip-bottom preview__in-collection"
-                ng:if="ctrl.showCollections && ctrl.image.data.collections.length > 0">
-                <gr-icon gr-small>folder</gr-icon>
-                <span class="titip-content preview__in-collection__list">
-                    <li ng-repeat="collection in ctrl.image.data.collections">
-                        <a ui:sref="search.results({query: '~'+collection.data.pathId})">
-                            {{collection.data.path.join(' ▸ ')}}
-                        </a>
-                        <br>
-                    </li>
-                </span>
-            </span>
+            <ul class="bottom-bar__meta-item titip-bottom preview__collections">
+                <li class="preview__collections__collection"
+                    ng:repeat="collection in ctrl.image.data.collections"
+                    style="background-color: {{collection.data.cssColour}}">
+                    <a ui:sref="search.results({query: '~'+collection.pathId})"
```

Comment:
Like any expression in templates, it may be re-evaluated whenever a scope is digested. I am not aware of any difference in performance between `{{ transform(value) }}` and `{{ value | transform }}`? For performance, it's more important to mark immutable expressions as one-time-bind using `::`, so either `{{:: transform(value) }}` or `{{:: value | transform }}`, which again should only run once in both cases.

I haven't checked the uiRouter code, but I suspect that in the case above, the `uiSref` attribute is merely a `&` attribute, which only gets invoked (and therefore evaluated) when triggered, in this case on click. I might be wrong though as it depends on the implementation.

I agree there's a risk of slowing down performance with expensive filters, but I think it's important we all understand why and how Angular works to avoid superstition or false optimisations.